action.yml

```diff
@@ -36,4 +36,5 @@ runs:
     ARG_DEBUG: ${{ inputs.debug }}
   entrypoint: '/scripts/lock.sh'
   post-entrypoint: '/scripts/unlock.sh'
+  post-if: ${{ always() }}
```

Comment:
I'm pretty sure this is already the default. The current behaviour is already "if a step fails, the lock gets cleaned up".

Reply:
Here's the doc: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runspost-if

Reply:
I think the difference is in “canceling” vs “failing”. When we cancel a job I’m pretty sure the whole pipeline just gets terminated. When a step fails the pipeline keeps running, which is when the post jobs are triggered